ed448-goldilocks/src/montgomery/x.rs

```diff
@@ -0,0 +1,457 @@
+// use crate::constants::A_PLUS_TWO_OVER_FOUR;
+use super::{
+    DEFAULT_ENCODE_TO_CURVE_SUITE, DEFAULT_HASH_TO_CURVE_SUITE, MontgomeryPoint, MontgomeryScalar,
+    ProjectiveMontgomeryPoint,
+};
+use crate::field::{ConstMontyType, FieldElement, FieldElementU84};
+use crate::{AffinePoint, Curve448};
+use core::fmt;
+use core::ops::Mul;
+use elliptic_curve::array::Array;
+use elliptic_curve::bigint::U448;
+use elliptic_curve::consts::{U28, U84};
+use hash2curve::{ExpandMsg, ExpandMsgXof, Expander, FromOkm, MapToCurve};
+use sha3::Shake256;
+use subtle::{Choice, ConditionallyNegatable, ConditionallySelectable, ConstantTimeEq};
+
+// Low order points on Curve448 and it's twist
+const LOW_A: MontgomeryXpoint = MontgomeryXpoint([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+]);
+const LOW_B: MontgomeryXpoint = MontgomeryXpoint([
+    0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+]);
+const LOW_C: MontgomeryXpoint = MontgomeryXpoint([
+    0xfe, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xfe, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+]);
+
+/// A point in Montgomery form
+#[derive(Copy, Clone)]
+pub struct MontgomeryXpoint(pub [u8; 56]);
```

Comment:
Pendantic question: shouldn't that be named `MontgomeryXPoint` to abide by https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#general-conventions-[rfc-#430] ?

```suggestion
pub struct MontgomeryXPoint(pub [u8; 56]);
```

Reply:
Yep, Rust API guidelines is also what I followed in this case.